### ISSUE TITLE
EmitBatchUsageEventAsync fails with JsonException

### ISSUE THREAD

**Issue by lostmsu** [Azure/Commercial-Marketplace-SaaS-Accelerator#100]
**Describe the bug**
The following code produces `System.Text.Json.JsonException`: _The input does not contain any JSON tokens_.
```csharp
var bill = new[] {
    new MeteringUsageRequest {
        Dimension = "test-cpu",
        EffectiveStartTime = new DateTime(target.Hour, DateTimeKind.Utc),
        Quantity = count,
        PlanId = subscription.AmpplanId,
        ResourceId = target.SubscriptionID,
    },
    new MeteringUsageRequest {
        Dimension = "test-gpu",
        EffectiveStartTime = new DateTime(target.Hour, DateTimeKind.Utc),
        Quantity = count/2.0,
        PlanId = subscription.AmpplanId,
        ResourceId = target.SubscriptionID,
    },
};

var response = await this.billingClient.EmitBatchUsageEventAsync(bill);
```

**Stack Trace**

```
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& readStack, JsonReaderException ex)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.Deserialize(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at Microsoft.Marketplace.SaasKit.Network.MeteringApiRestClient`1.ProcessErrorResponse(String url, WebException ex) in D:\a\1\s\src\SaaS.SDK.Client\Network\MeteringApiRestClient.cs:line 52
   at Microsoft.Marketplace.SaasKit.Network.AbstractSaaSApiRestClient`1.DoRequest(String url, String method, Dictionary`2 parameters, Dictionary`2 headers, String contentType) in D:\a\1\s\src\SaaS.SDK.Client\Network\AbstractSaaSApiRestClient.cs:line 78
   at Microsoft.Marketplace.SaasKit.Services.MeteredBillingApiClient.EmitBatchUsageEventAsync(IEnumerable`1 subscriptionBatchUsageRequest) in D:\a\1\s\src\SaaS.SDK.Client\Services\MeteredBillingAPIClient.cs:line 95
   at LostTech.Licensing.MeteringCollector.BillViaMeteringApi.Run(Target target, CloudTable reportsTable, ILogger log) in D:\a\1\s\src\Azure.MeteringCollector\BillViaMeteringApi.cs:line 82
```

**Expected behavior**
The exception should explain the reason of the call failure. Instead, SDK fails to parse response.

**Environment:**
Azure Functions v3

**Remarks**

```ProcessErrorResponse``` appears to completely ignore exceptions it does not know about, and just silently returns `null`, which seems to be dangerous.

**Comment by lostmsu** [comment]
OK, I've added some logging to my fork:

```csharp
try {
    meteredBillingErrorResult = JsonSerializer.Deserialize<MeteringErrorResult>(responseString);
} catch (JsonException) {
    meteredBillingErrorResult = new MeteringErrorResult {ErrorMessage = responseString};
}
```

Turns out I received HTTP 404:
> Unable to find the request https://marketplaceapi.microsoft.com/api/batchUsageEvent?api-version=2018-08-31

`responseString` is empty.

**Comment by santhoshb-msft** [comment]
Hi @lostmsu ,
You should still be able to test even if the offer is in "Preview". Could you postman the same call, see if it works?

**Comment by lostmsu** [comment]
@santhoshb-msft I figured out my problem: I missed `SaaSApiConfiguration__TenantId` in my function ARM template.

The original issue thus was caused by `ADAuthenticationHelper.GetAccessToken` failing with `WebException` in `DoRequest`, leading to a confusion, as `DoRequest` reports the error as if it happened during the call to metering API itself. I was able to clear it after splitting the `try` block in `DoRequest` into two: one for `GetAccessToken` and other for the actual request submission.
